src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb

```diff
@@ -104,33 +104,50 @@ def create_stemcell(image_path, cloud_properties)
         begin
           Dir.mktmpdir do |tmp_dir|
             @logger.info('Creating new image...')
+
+            is_glance_v1 = @openstack.image.class.to_s.include?('Fog::Image::OpenStack::V1')
+
             image_params = {
-              :name => "BOSH-#{generate_unique_name}",
+              :name => "#{cloud_properties['name']}/#{cloud_properties['version']}",
               :disk_format => cloud_properties['disk_format'],
               :container_format => cloud_properties['container_format'],
-              :is_public => @stemcell_public_visibility.nil? ? false : @stemcell_public_visibility,
             }
 
+            is_public = !@stemcell_public_visibility.nil? && @stemcell_public_visibility.to_s != 'false'
+            if is_glance_v1
+              image_params[:is_public] = is_public.to_s
```

Comment:
@beyhan good call, not necessary. Fixed in https://github.com/cloudfoundry-incubator/bosh-openstack-cpi-release/pull/49/commits/86c4881fbbe1cb1540ba650d96c17d3736460e0b